lnd_test.go

```diff
@@ -1445,57 +1446,86 @@ func testUpdateChannelPolicy(net *lntest.NetworkHarness, t *harnessTest) {
 	closeChannelAndAssert(ctxt, t, net, net.Alice, chanPoint3, false)
 }
 
+// waitForNodeBlockHeight queries the node for its current block height until
+// it reaches the passed height.
+func waitForNodeBlockHeight(ctx context.Context, node *lntest.HarnessNode,
+	height int32) error {
+	var predErr error
+	err := lntest.WaitPredicate(func() bool {
+		ctxt, _ := context.WithTimeout(ctx, 10*time.Second)
+		info, err := node.GetInfo(ctxt, &lnrpc.GetInfoRequest{})
+		if err != nil {
+			predErr = err
+			return false
+		}
+
+		if int32(info.BlockHeight) != height {
+			predErr = fmt.Errorf("expected block height to "+
+				"be %v, was %v", height, info.BlockHeight)
+			return false
+		}
+		return true
+	}, 15*time.Second)
+	if err != nil {
+		return predErr
+	}
+	return nil
+}
+
 // testOpenChannelAfterReorg tests that in the case where we have an open
 // channel where the funding tx gets reorged out, the channel will no
 // longer be present in the node's routing table.
 func testOpenChannelAfterReorg(net *lntest.NetworkHarness, t *harnessTest) {
-	ctxb := context.Background()
+	var (
+		ctxb = context.Background()
+		temp = "temp"
+		perm = "perm"
+	)
 
 	// Set up a new miner that we can use to cause a reorg.
 	args := []string{"--rejectnonstd", "--txindex"}
-	miner, err := rpctest.New(harnessNetParams,
+	tempMiner, err := rpctest.New(harnessNetParams,
 		&rpcclient.NotificationHandlers{}, args)
 	if err != nil {
 		t.Fatalf("unable to create mining node: %v", err)
 	}
-	if err := miner.SetUp(true, 50); err != nil {
+	if err := tempMiner.SetUp(true, 50); err != nil {
```

Comment:
It doesn't seem this actually needs to be true? As is it'll just go through an unnecessary reorg when attempting to connect it to the existing miner with a longer chain to reorg it once again later on. Maybe we don't gain much time saved after all anyway.

Reply:
Yep!